cardano-api/src/Cardano/Api/KeysByron.hs

```diff
@@ -54,10 +72,21 @@ import           Cardano.Api.SerialiseTextEnvelope
 -- This is a type level tag, used with other interfaces like 'Key'.
 --
 data ByronKey
+data ByronKeyLegacy
 
-instance HasTypeProxy ByronKey where
-    data AsType ByronKey = AsByronKey
-    proxyToAsType _ = AsByronKey
+class IsByronKey key where
+    byronKeyFormat :: ByronKeyFormat key
+
+data ByronKeyFormat key where
+  ByronLegacyKeyFormat :: ByronKeyFormat ByronKeyLegacy
+  ByronNonLegacyKeyFormat :: ByronKeyFormat ByronKey
+
+data ByronWitness = LegacyWitness (SigningKey ByronKeyLegacy)
+                  | NonLegacyWitness (SigningKey ByronKey)
```

Comment:
I think we should call this `SomeByronSigningKey`, to match the `SomeSigningKey` we have in the CLI.

It's not a witness, it's just a signing key. (Yes signing keys are used to construct witnesses, but they're not a witness)

How about:
```
data SomeByronSigningKey = AByronSigningKey (SigningKey ByronKey)
                         | AByronSigningKeyLegacy (SigningKey ByronKeyLegacy)
```